### PR TITLE
Use xmllint step of a2x for patching manpages

### PR DESCRIPTION
**Pull request by t-wissmann**
This essentially reverts the cmake changes of f61211ad0f (#1396) and
instead injects the patch-manpage-xml.py via asciidoc's xmllint step.

The positive effect of this is that we do not need to know the path used
in ASCIIDOC_MANPAGE_XSL. Recently on arch linux, it was moved from

    /etc/asciidoc/docbook-xsl/manpage.xsl

to

    /usr/lib/python3.10/site-packages/asciidoc/resources/docbook-xsl/manpage.xsl

which of course depends on a lot of parameters (python version, etc).
Since I couldn't find a way to let asciidoc tell me the path, so having
the path a cmake parameter will presumably cause a lot of packaging
headache.

Instead, I noticed, that we could abuse the 'xmllint' step of the
asciidoc compilation to inject a tool that patches the man page xml.
As a side-effect, this changes make the documentation build again
successfully out of the box on arch linux.

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -75,34 +74,22 @@ function(gen_manpage sourcefile man_nr)
...
+        "XMLLINT = '${CMAKE_CURRENT_SOURCE_DIR}/patch-manpage-xml.py'")
         add_custom_command(
-            OUTPUT ${dst} ${docbookxml}
-            # first create the docbook xml:
-            COMMAND ${ASCIIDOC}
-                    -o ${docbookxml}
-                    -b docbook45
+            OUTPUT ${dst}
+            COMMAND ${ASCIIDOC_A2X}
+                    --verbose
```
Comment (blarz) [inline doc/CMakeLists.txt:85]: Do we still want this or was this just for debugging purposes?

Comment (t-wissmann) [reply doc/CMakeLists.txt:85]: I'd like to keep it because otherwise, a2x shows too little information.

Comment (blarz) [reply doc/CMakeLists.txt:85]: Alright! Just wanted to make sure this is intended :)

**Review by blarz — APPROVED** [review]
Nice workaround :)

**Comment by t-wissmann** [comment]
Thanks for reviewing (this and the other PR) :-)

**Pull request merged**